nac_test/robot_writer.py

```diff
@@ -118,7 +143,71 @@ def _fix_duplicate_path(self, *paths: str) -> Path:
                 return Path(*paths[:-1], "_" + paths[-1])
         return Path(os.path.join(*paths))
 
-    def write(self, templates_path: Path, output_path: Path) -> None:
+    @staticmethod
+    def _calculate_full_suite_name(output_path: Path, robot_file: Path) -> str:
+        """
+        We need to collect the final robot suite name (ex. Output.Config.Tenants.L3Out)
+        and note this in the ordering file. The suite name is derived from
+        1. the output path (the last part of it if the path is an absolute path)
+        2. the path of the robot file relative the the output path
+        Each part of 1 and 2 is passed through a robot API to (ex: config -> Config) and
+        joined with a dot (.) to form the suite name.
+        Example 1:
+            Input:  output_path = /tmp/foo/output
+                    filename = /tmp/foo/output/config/tenants/ABC/L3Out.robot
+            Result: suite_dirname = Output.Config.Tenants.ABC.L3Out
+        Example 2:
+            Input:  output_path = foobar
+                    filename = foobar/integration_tests/whatever.robot
+            Result: suite_dirname = Foobar.Integration Tests.Whatever
+        """
+        relative_path = robot_file.parent.relative_to(output_path)
+        path_parts = [output_path.name] + list(relative_path.parts) + [robot_file.stem]
+        return ".".join([TestSuite.name_from_source(p) for p in path_parts if p])
+
+    def _update_ordering_entries(
+        self, output_path: Path, robot_file: Path, delete_empty_suite: bool = False
```

Comment:
Do we need "delete_empty_suite" if we always set it to "true"?

Reply:
I just pushed a diff to remove this logic.. too dangerous if you ask me..